spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/StreamListenerResultAdapter.java

```diff
@@ -41,6 +43,6 @@
 	 * @param streamListenerResult the result of invoking the method.
 	 * @param bindingTarget the binding target.
 	 */
-	void adapt(R streamListenerResult, B bindingTarget);
+	Closeable adapt(R streamListenerResult, B bindingTarget);
```

Comment:
This is breaking change in the minor version.
I understand that you have an argument like impls are an internal stuff and it is unusual to have something custom in the target app and so on.
But would be better to keep that in mind for similar changes in the future.
Only major version can afford such a squander.